examples/bench/main.go

```diff
@@ -0,0 +1,435 @@
+// Copyright 2018 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"image"
+	"image/color"
+	_ "image/png"
+	"log"
+	"os"
+	"runtime/pprof"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/examples/resources/images"
+	"github.com/hajimehoshi/ebiten/internal/testflock"
+)
+
+var noErr = errors.New("clean")
+
+type benchmark struct {
+	name   string
+	fn     func(b *testing.B, screen *ebiten.Image)
+	result testing.BenchmarkResult
+	output []string
+}
+
+const (
+	MAX_TRIANGLES = 1000
+	TRIANGLE_SIZE = 20
+)
+
+var (
+	width  = 640
+	height = 480
+	// images to use during the benchmarks
+	px26  *ebiten.Image
+	px104 *ebiten.Image
+	px416 *ebiten.Image
+	box   *ebiten.Image
+
+	benchIdx    = 0
+	results     []testing.BenchmarkResult
+	setup       sync.Once
+	benchFunc   func(b *testing.B, screen *ebiten.Image)
+	benchGo     = make(chan struct{})
+	benchDone   = make(chan bool)
+	vertices    = make([]ebiten.Vertex, 0, MAX_TRIANGLES*3)
+	indices     = make([]uint16, 0, MAX_TRIANGLES*3)
+	rainbow     = make([][3]float32, 120)
+	rainbowBase = []color.RGBA{
+		{240, 0, 0, 255},
+		{240, 90, 0, 255},
+		{220, 220, 0, 255},
+		{0, 200, 0, 255},
+		{0, 0, 255, 255},
+		{180, 0, 200, 255},
+	}
+)
+
+func main() {
```

Comment:
If this is a set of benchrmark, would `TestMain` in a test package be better? I was wondering if there was a reason you wrote this as a main package.